source/adapters/hip/enqueue.cpp

```diff
@@ -313,19 +312,15 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
 
   // Early exit for zero size range kernel
   if (*pGlobalWorkSize == 0) {
-    if (DepEvents.size()) {
-      return urEnqueueEventsWaitWithBarrier(hQueue, DepEvents.size(),
-                                            phEventWaitList, phEvent);
-    }
-    return UR_RESULT_SUCCESS;
+    return urEnqueueEventsWaitWithBarrier(hQueue, numEventsInWaitList,
+                                          phEventWaitList, phEvent);
```

Comment:
This will enqueue a barrier all the time not just when there's `DepEvents`, the `enqueueEventsWait` helper will skip if there's no events but I don't think this will.

Reply:
This is only true if `*pGlobalSize` is zero. This is the correct behaviour. Execution should not proceed past this point if there is no global size. Checking for size previously was a bug

Reply:
Oh I see that's what the CUDA adapter is doing, and I guess without it you might end up with return event not set